### DISCUSSION TITLE
Custom OPENID_CALLBACK_URL results in Unexpected Application Error (404)

### DISCUSSION THREAD

**Discussion by fbaeumer** [danny-avila/LibreChat#3199] [Troubleshooting] [answered]
Hello everyone,

We are currently implementing an SSO with OpenID. We have the situation that the login page of the SSO service is loaded, where you can also log in. The redirect should then actually take place. However, only an empty error page appears in LibreChat. I think it could be due to the custom OPENID_CALLBACK_URL. Unfortunately, there is no other error message and no log entry. It is very difficult to determine what the problem might be. 

<img width="260" alt="image" src="https://github.com/danny-avila/LibreChat/assets/25347417/d0f064db-6383-4151-bbb0-85519f6325c8">

The resulting URL

https://chat.xxxx.nl/oauth/**sign**/callback?state=najcpDNfkRJAT58bE-4MIM-pfXbTbdVET49OO_lPhBUs&code=ba7a469517e46k858c32a636b2e82c292bbf420190bff860a7dfb15e139874cec&session_state=11eK%2BHcfBt2hkEQd8r2hidOW%2FVBWndTScjrC2kOLRq%2FeY%3D.T0NKdWthOXIzREVES2E1MCtkFYnhLb0NKUzVMbzVZSHVGNjc0UWhGaGUyVmkVTcXJiZFB2OVVBM2VkV3g1N1N6UjI0emRIUDVQcXltRHlONkVtcXJWVUE9PQ

Looks finde and everything URL-Related is also working. The config is

DOMAIN_CLIENT=https://chat.xxxx.nl
DOMAIN_SERVER=https://chat.xxxx.nl

# OpenID
OPENID_CLIENT_ID=chat.xxxx.nl
OPENID_CLIENT_SECRET=a8adjUda77sssda2e
OPENID_ISSUER=https://auth-service.nl
OPENID_SESSION_SECRET=eassadsadsa
OPENID_SCOPE="openid profile email"
OPENID_CALLBACK_URL=/oauth/sign/callback
#OPENID_REQUIRED_ROLE=
#OPENID_REQUIRED_ROLE_TOKEN_KIND=
#OPENID_REQUIRED_ROLE_PARAMETER_PATH=

OPENID_BUTTON_LABEL="Continue as XXXXX User"
#OPENID_IMAGE_URL=

Any idea what to check next? We can't change the URL since its given.

Thanks

**Comment by danny-avila** [comment, marked as answer]
> I think it could be due to the custom OPENID_CALLBACK_URL

Yes, you would need to edit the code and add the custom path in `api/server/routes/oauth.js` in order to use a custom `OPENID_CALLBACK_URL` (with the "openid" part replaced for "sign")

I think the intention behind making it configurable was to add query params, I'm not sure
